### DISCUSSION TITLE
How can automa fill value to a form in iframe?

### DISCUSSION THREAD

**Discussion by kingname** [AutomaApp/automa#1772] [Q&A]
According to the Same-Origin Policy, if A page use iframe to load B page, we can not use Javascript in A page to fill the form in B page if they have different domain. But I find that Automa can do it. That is awesome and excellent. How can automa do it? Please tell me which file has this magic. I'd like to study. Thanks.

**Comment by jingbof** [comment]
Automa can interact with iframes across different domains because it's a browser extension with elevated permissions. It injects content scripts into both the main page and the iframe, allowing it to bypass the Same-Origin Policy that normally applies to regular web page scripts. Basically three things:

1. extension privileges
2. content script injection
3. background script & messaging

**Reply by qkxie** [reply]
Sure. After this question in June, I have found the solution. The important points are messaging and the injected script in the iframed page.
